File: src/main/cpp/subsystems/intake/Intake.cpp
#include "subsystems/intake/Intake.h"


Intake::Intake() {
    motor.SetSmartCurrentLimit(
            CONSTANTS::INTAKE::MOTOR::HARD_CURRENT_LIMIT.to<double>());
    motor.SetSecondaryCurrentLimit(
            CONSTANTS::INTAKE::MOTOR::SOFT_CURRENT_LIMIT.to<double>());
    motor.SetIdleMode(rev::CANSparkMax::IdleMode::kBrake);
    motor.SetInverted(true);
}

units::degree_t Intake::GetAngle() {
    return rotation.Get();
}

void Intake::SetAngle(units::degree_t angle) {
    rotation.Set(angle);
}

double Intake::GetMotor() {
    return motor.Get();
}

void Intake::SetMotor(double power) {
    motor.Set(power);
}

void Intake::ZeroEncoder(units::degree_t angle) {
    rotation.Zero(angle);
}

File: src/main/java/frc/bionicpolars/subsystems/intake/Intake.java
package frc.bionicpolars.subsystems.intake;

import com.revrobotics.CANSparkMax;

import edu.wpi.first.wpilibj2.command.SubsystemBase;
import frc.bionicpolars.Constants;

public class Intake extends SubsystemBase {
    CANSparkMax motor = new CANSparkMax(Constants.INTAKE_MOTOR_ID,
            CANSparkMax.MotorType.kBrushless);
    IntakeRotate rotation = new IntakeRotate();

    public double getAngle() {
        return rotation.Get();
    }

    public void setAngle(double angle) {
        rotation.Set(angle);
    }

    public double getMotor() {
        return motor.get();
    }

    public void setMotor(double speed) {
        motor.set(speed);
    }

    public void zeroEncoder(double angle) {
        rotation.Zero(angle);
    }
}
